controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +74,37 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator sslCon = new SSLContextConfigurator();
+                sslCon.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                sslCon.setKeyStorePass(loadPasswordFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(sslCon, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
         } finally {
             LoggerHelpers.traceLeave(log, this.objectId, "startUp", traceId);
         }
     }
 
+    private String loadPasswordFromFile(String keyFilePasswordPath) {
+
+        if (Strings.isNullOrEmpty(keyFilePasswordPath)) {
+            return "";
```

Comment:
Is it correct to return empty string? Shouldn't it the be an error? I'd like to understand the consequence of returning an empty string here.

Reply:
Empty string means the password is not used. We return empty string if user have not specified the password path. Which means that user is not planning to use a password for the keyfile. Added a comment to the code.